### DISCUSSION TITLE
Does GeoJsonLayer support BitmapLayer masking?

### DISCUSSION THREAD

**Discussion by tangzijun** [visgl/deck.gl#8105] [Q&A]
I found that this has no effect. Is there something wrong with my writing? Or does it not support this itself?

My GeoJson is the boundary information of a country, and the Bitmap is a picture.

```
export class TempLayer extends CompositeLayer {
  renderLayers() {
    return [
      new GeoJsonLayer({
        id: 'maskLayer',
        data: xingzhengquHelper.country(),
        filled: true,
        lineWidthMinPixels: 2,
        getFillColor: () => [255, 255, 255, 255],
        getLineColor: () => [255, 255, 255, 255],
        operation: 'mask',
      }),
      new BitmapLayer({
        id: 'temp-layer',
        image: '/weather-data/temp.png',
        bounds: [70.0, 0.0, 140.0, 60.0],
        opacity: 0.2,
        extensions: [new MaskExtension()],
        maskId: 'maskLayer',
      }),
    ];
  }
}
```

**Comment by tangzijun** [comment]
This error seems to only occur in CompositeLayer. When placing 2 layers into DeckGL layers, it works normally.

**Reply by Pessimistress** [reply]
It may or may not be related, but you shouldn't hard code sublayer ids. If you have multiple TempLayers in your app then it will create a id collision.
